staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go

```diff
@@ -643,6 +643,15 @@ func TestValidationExpressions(t *testing.T) {
 				"size(self.val) == 2",
 				"self.val.map(k, self.val[k]).exists(v, v == 1)",
 				"size(self.val.filter(k, self.val[k] > 1)) == 1",
+
+				// two variable comprehensions
+				"self.val.all(k, v, v > 0)",
+				"self.val.exists(k, v, v == 2)",
+				"self.val.existsOne(k, v, v == 2)",
+				"self.val.transformMap(k, v, v > 1, v + 1).size() == 1",
+				"self.val.transformMap(k, v, v + 1).size() == 2",
+				"self.val.transformMapEntry(k, v, v > 1, {k + '2': v + 1}).size() == 1",
+				"self.val.transformMapEntry(k, v, {k + '2': v + 1}).size() == 2",
```

Comment:
trying to understand the cost / power of the `transform*` functions
- do they mutate in place, shallow-copy, or deep-copy the input?
- do they only work on literal lists / maps or would they also work on fields in things like CRD validation / admission?

Reply:
They're macros. They expand to a common fold function (plus whatever other operations need to be composed with the fold to get the needed traversal/accumulation).  Since we already have cost to the functions the macro expands to, we're all set.

xref: https://github.com/google/cel-go/blob/8ad600b649be1b9ef5a003e8c5632d89b9aaf790/parser/macro.go#L213

Reply:
The fact that our costs noticed `@not_strictly_false` give me some added confidence we're not letting things slip through.